src/cascadia/TerminalControl/SearchBoxControl.cpp

```diff
@@ -0,0 +1,185 @@
+﻿// Copyright (c) Microsoft Corporation
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "SearchBoxControl.h"
+#include "SearchBoxControl.g.cpp"
+
+using namespace winrt;
+using namespace winrt::Windows::UI::Xaml;
+using namespace winrt::Windows::UI::Core;
+
+namespace winrt::Microsoft::Terminal::TerminalControl::implementation
+{
+    // Constructor
+    SearchBoxControl::SearchBoxControl() :
+        _goForward(false)
+    {
+        InitializeComponent();
+
+        this->CharacterReceived({ this, &SearchBoxControl::_CharacterHandler });
+
+        _textBox = this->FindName(L"TextBox").try_as<Controls::TextBox>();
+
+        if (_textBox)
+            _focusableElements.insert(_textBox);
+        _focusableElements.insert(CloseButton());
+        _focusableElements.insert(CaseSensitivityButton());
+        _focusableElements.insert(GoForwardButton());
+        _focusableElements.insert(GoBackwardButton());
+    }
+
+    // Method Description:
+    // - Getter for _goForward
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - bool: the value of _goForward
+    bool SearchBoxControl::GoForward()
+    {
+        return _goForward;
+    }
+
+    // Method Description:
+    // - Get the current state of the case button
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - bool: whether the case button is checked (sensitive)
+    //   or not
+    bool SearchBoxControl::IsCaseSensitive()
+    {
+        return _caseButton.IsChecked().GetBoolean();
+    }
+
+    // Method Description:
+    // - Handler for pressing Enter on TextBox, trigger
+    //   text search
+    // Arguments:
+    // - sender: not used
+    // - e: event data
+    // Return Value:
+    // - <none>
+    void SearchBoxControl::TextBoxKeyDown(winrt::Windows::Foundation::IInspectable const& /*sender*/, Input::KeyRoutedEventArgs const& e)
+    {
+        if (e.OriginalKey() == winrt::Windows::System::VirtualKey::Enter)
+        {
+            auto const state = CoreWindow::GetForCurrentThread().GetKeyState(winrt::Windows::System::VirtualKey::Shift);
+            if (WI_IsFlagSet(state, CoreVirtualKeyStates::Down))
+            {
+                // We do not want the direction flag to change permanately
+                _goForward = !_goForward;
+                _SearchHandlers(*this, _textBox.Text());
+                _goForward = !_goForward;
+            }
+            else
+            {
+                _SearchHandlers(*this, _textBox.Text());
+            }
+        }

```

Comment:
You'll probably want to mark the event as handled here too, at least for Enter and Escape